beginner_source/dist_overview.rst

```diff
@@ -15,107 +15,62 @@ to the technology that can best serve your use case.
 Introduction
 ------------
 
-As of PyTorch v1.6.0, features in ``torch.distributed`` can be categorized into
-three main components:
-
-* `Distributed Data-Parallel Training <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
-  (DDP) is a widely adopted single-program multiple-data training paradigm. With
-  DDP, the model is replicated on every process, and every model replica will be
-  fed with a different set of input data samples. DDP takes care of gradient
-  communication to keep model replicas synchronized and overlaps it with the
-  gradient computations to speed up training.
-* `RPC-Based Distributed Training <https://pytorch.org/docs/stable/rpc.html>`__
-  (RPC) supports general training structures that cannot fit into
-  data-parallel training such as distributed pipeline parallelism, parameter
-  server paradigm, and combinations of DDP with other training paradigms. It
-  helps manage remote object lifetime and extends the
-  `autograd engine <https://pytorch.org/docs/stable/autograd.html>`__ beyond
-  machine boundaries.
-* `Collective Communication <https://pytorch.org/docs/stable/distributed.html>`__
-  (c10d) library supports sending tensors across processes within a group. It
+The PyTorch Distributed library includes a collective of parallelism modules,
+a communications layer (C10D), and infrastructure for launching and
+debugging large training jobs.
+
+These Parallelism Modules offer high-level functionality and compose with existing models:
+* `Distributed Data-Parallel (DDP) <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
+* `Fully Sharded Data-Parallel Training (FSDP) <https://pytorch.org/docs/stable/fsdp.html>`__
+* TODO what to mention about FSDP2 vs FSDP
+* `Tensor Parallel <https://pytorch.org/docs/stable/distributed.tensor.parallel.html>`__
+* TODO link to pipeline parallel after stable docs link exists
+
+DTensor and DeviceMesh are primitives used to build parallelism in terms of sharded or replicated tensors on N-dimensional process groups.
```

Comment:
Should those be in backticks as names of modules? if yes, please wrap them, otherwise add to en-wordlist.txt to pass spellcheck. 

Please add `reshard` and `sharded` to en-wordlist.txt.

Reply:
DTensor is a class name, not a module name.  Should it be added with backticks too?

If i need to add DTensor to the spelling wordlist, do i need to match the capitalization, or can i just put lowercase dtensor in there and DTensor would also match?

Reply:
Yes, backticks would work for a class as well. Intersphinx should work too if you'd prefer that.
Both DTensor and dtensor would need to be added. But I'd prefer backticks =) as that's the correct formatting.

Reply:
hmm, the backticks worked for DTensor/DeviceMesh, but ProcessGroups rendered badly, like  \``ProcessGroup``s